crates/rattler_networking/src/authentication_middleware.rs

```diff
@@ -72,8 +74,39 @@ impl AuthenticationMiddleware {
         }
     }
 
+    ///Authenticate with Google Cloud
+    async fn authenticate_with_google_cloud(
+        mut req: reqwest::Request,
+    ) -> reqwest_middleware::Result<reqwest::Request> {
+        let audience = "https://storage.googleapis.com/";
+        let scopes = [
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/devstorage.full_control",
```

Comment:
Is this the lowest permission? I don't think we need full control all the time, right? Maybe read-only is possible?